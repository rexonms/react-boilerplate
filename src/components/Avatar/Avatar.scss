.container {
  width: 40px;
  height: 40px;
  border-radius: 20px;
  overflow: hidden;

  img {
    width: 100%;
    height: 100%;
  }
}