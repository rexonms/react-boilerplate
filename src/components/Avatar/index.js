import React from 'react';
import PropTypes from 'prop-types';

import styles from './Avatar.scss';

const Avatar = props => (
  <div className={`${styles.container}`}>
    <img src={props.imageURL} />
  </div>
);

Avatar.defaultProps = {};

Avatar.propTypes = {
  /** Url of the image */
  imageURL: PropTypes.string.isRequired,
};

export default Avatar;
