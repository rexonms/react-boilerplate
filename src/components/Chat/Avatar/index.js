import React from 'react';
import PropTypes from 'prop-types';

import styles from './Avatar.scss';

const propTypes = {
  /** URL of the image to be displayed */
  imageURL: PropTypes.string.isRequired,
  /** Alt text for image */
  imageAltText: PropTypes.string.isRequired,
};

const defaultProps = {};

const Avatar = props => (
  <div className={`${styles.container}`}>
    <img src={props.imageURL} alt={props.imageAltText} />
  </div>
);

Avatar.defaultProps = defaultProps;
Avatar.propTypes = propTypes;

export default Avatar;
