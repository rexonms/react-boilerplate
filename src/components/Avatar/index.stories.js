import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import Avatar from './index';

const stories = storiesOf('Components/Avatar', module);
stories.addDecorator(withKnobs);

// Knobs as dynamic variables.
stories.addDecorator(story => (story()));

stories.add('Avatar', withInfo(`
  The avatar of the user
  ~~~js
    <Avatar imageURL={imageURL} />)
  ~~~
`)(() => {
  return (
    <Avatar imageURL='https://place-hold.it/100' />);
}));

