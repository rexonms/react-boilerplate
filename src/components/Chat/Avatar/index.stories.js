import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import Avatar from './index';

const stories = storiesOf('components/Chat/Avatar', module);
const imageUrl = 'https://place-hold.it/100'
stories.addDecorator(withKnobs);
stories.addDecorator(story => (story()));
stories.add('Avatar', withInfo(`
  Component
  ~~~js
    <Avatar imageURL=imageUrl imageAltText="imageAltText"/>)
  ~~~
`)(() => {
  return (
    <Avatar
      imageURL={imageUrl}
      imageAltText="imageAltText"
    />
  );
}));
