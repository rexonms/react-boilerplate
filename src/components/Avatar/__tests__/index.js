import React from 'react';
import renderer from 'react-test-renderer';

import Avatar from './../index';

it('renders correctly', () => {
  const tree = renderer.create(<Avatar imageURL='https://place-hold.it/100' />).toJSON();
  expect(tree).toMatchSnapshot();
});
