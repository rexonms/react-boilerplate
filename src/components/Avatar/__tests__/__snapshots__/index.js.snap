// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <img
    src="https://place-hold.it/100"
  />
</div>
`;
